pytensor/tensor/rewriting/linalg.py

```diff
@@ -611,3 +613,20 @@ def rewrite_inv_inv(fgraph, node):
     ):
         return None
     return [potential_inner_inv.inputs[0]]
+
+
+@register_canonicalize
+@register_stabilize
+@node_rewriter([ExtractDiag])
+def rewrite_diag_kronecker(fgraph, node):
```

Comment:
What about diagonals that result from multiplication with eye?

Reply:
Also needs a docstring

Reply:
this rewrite does not deal with the creation of a diagonal matrix. the input should be the kron product and diag is the outer op